File: py_examples/spawn_2_seq.py
"""
spawn_2.py : Runs two CPU bound functions sequentially

Daniel Zahka ~ daniel.zahka@wustl.edu
Anish Naik ~ anish.r.naik@wustl.edu
"""

def thread_func():
        for i in xrange(1, 50000000):
                k = i * 3


if __name__ == "__main__":
    print("Starting work...")
    thread_func()
    thread_func()
    print("Done...")

File: C_Examples/spawn_2_seq.c
/**
 * Spawn_2_seq.c : A program to run two cpu bound functions
 * sequentially
 *
 * Daniel Zahka ~ daniel.zahka@wustl.edu
 * Anish Naik ~ anish,r.naik@wustl.edu
 *
 */

#include <stdlib.h>
#include <pthread.h>
#include <stdio.h>

#define ITERATIONS (1000000000)

void * thread_func(void * args){
    long long i, k;
    for (i = 0; i < ITERATIONS; ++i){
        k = i * 3;
    }
    return 0;
}


int main(){
    printf("Spawning Threads\n");
    thread_func(0);
	thread_func(0);
	printf("Joined Threads\n");
 
    return 0;
}
